### PR TITLE
Use side type everywhere

### PR DESCRIPTION
**Pull request by blair**

### CODE REVIEW

File: src/exchanges/gdax/GDAXExchangeAPI.ts [hunk 1]
```diff
@@ -199,15 +198,13 @@ export class GDAXExchangeAPI implements PublicExchangeAPI, AuthenticatedExchange
             return Promise.reject(new GTTError('No authentication details were given for this API'));
         }
         let gdaxOrder: OrderParams;
-        assert(order.side === 'buy' || order.side === 'sell');
```
Comment (fb55) [inline src/exchanges/gdax/GDAXExchangeAPI.ts]: Why was the runtime check removed?

Comment (blair) [reply src/exchanges/gdax/GDAXExchangeAPI.ts]: @fb55 For consistency with the rest of the code, there are no runtime checks anywhere. Plus, PlaceOrderMessage above change's the 'side' property from a string to a Side, so there should be a compile time check.

Comment (fb55) [reply src/exchanges/gdax/GDAXExchangeAPI.ts]: Sg 👍

**Comment by fb55** [comment]
The rebase looks a bit more involved :/

**Comment by blair** [comment]
@fb55 I'll rebase any PRs that don't merge cleanly and do any conflict resolution. I'll ping you when they are done.

**Comment by blair** [comment]
@fb55 Rebase completed. PTAL.

**Comment by fb55** [comment]
Great change, thanks a lot!

**Pull request merged**